File: app/types/note.ts
export type NoteType = 'all' | 'memo' | 'todo' | 'foreshadow' | 'issue' | 'idea'
export type NoteStatus = 'all' | 'open' | 'doing' | 'done' | 'archived'
export type NotePriority = 'low' | 'normal' | 'high'

export interface CreativeNote {
    id: string
    projectId: string
    documentId?: string | null
    documentTitle?: string | null
    paragraphId?: string | null
    type: Exclude<NoteType, 'all'> | string
    title: string
    body: string
    status: Exclude<NoteStatus, 'all'> | string
    priority: NotePriority | string
    dueAt?: number | null
    createdAt: number
    updatedAt: number
}

export interface CreateNoteInput {
    projectId: string
    documentId?: string | null
    paragraphId?: string | null
    type: Exclude<NoteType, 'all'> | string
    title: string
    body?: string | null
    priority?: NotePriority | string | null
    dueAt?: number | null
}

export interface UpdateNoteInput {
    noteId: string
    documentId?: string | null
    paragraphId?: string | null
    type?: Exclude<NoteType, 'all'> | string
    title?: string
    body?: string
    status?: Exclude<NoteStatus, 'all'> | string
    priority?: NotePriority | string
    dueAt?: number | null
}

export interface ListNotesInput {
    projectId: string
    documentId?: string | null
    paragraphId?: string | null
    type?: NoteType | string | null
    status?: NoteStatus | string | null
}

export interface EditorParagraphNoteRequest {
    type: Exclude<NoteType, 'all'>
    paragraphId: string | null
    selectedText: string
}

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct NoteDto {
    pub id: String,
    pub project_id: String,
    pub document_id: Option<String>,
    pub paragraph_id: Option<String>,
    #[serde(rename = "type")]
    pub note_type: String,
    pub title: String,
    pub body: String,
    pub status: String,
    pub priority: String,
    pub due_at: Option<i64>,
    pub created_at: i64,
    pub updated_at: i64,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateNoteInput {
    pub project_id: String,
    pub document_id: Option<String>,
    pub paragraph_id: Option<String>,
    #[serde(rename = "type")]
    pub note_type: String,
    pub title: String,
    pub body: Option<String>,
    pub priority: Option<String>,
    pub due_at: Option<i64>,
}
